samples/amber.cc

```diff
@@ -102,6 +105,29 @@ const char kUsage[] = R"(Usage: amber [options] SCRIPT [SCRIPTS...]
   -h                        -- This help text.
 )";
 
+// Parses a decimal integer from the given string, and writes it to |retval|.
+// Returns true if parsing succeeded and consumed the whole string.
+static bool ParseOneInt(const char* str, int* retval) {
+  char trailing = 0;
+#if defined(_MSC_VER)
+  return sscanf_s(str, "%d%c", retval, &trailing, 1) == 1;
+#else
+  return std::sscanf(str, "%d%c", retval, &trailing) == 1;
+#endif
+}
+
+// Parses a decimal integer, then a period (.), then a decimal integer  from the
+// given string, and writes it to |retval|.  Returns true if parsing succeeded
+// and consumed the whole string.
+static int ParseIntDotInt(const char* str, int* retval0, int* retval1) {
+  char trailing = 0;
```

Comment:
What is trailing for? Do we not care if it's parsed or what it parses as? If so, why parse it?

Reply:
We attempt to parse it, and hope for failure.  It will soak up *any* extra text at the end, if it exists.

The return value on sscanf (and sscanf_s) is the number of successfully converted / parsed conversion-specs. So with %d.%d%c we hope for the %d and . and %d to be fulfilled and the %c to not be fulfilled.

Sorry, this is an idiom I'm used to.